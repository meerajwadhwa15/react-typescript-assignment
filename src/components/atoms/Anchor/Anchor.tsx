import React, { FunctionComponent, ReactNode } from 'react'
import { Link } from 'react-router-dom'

export type Props = {
  url: string
  children: ReactNode
  title?: string
}

const Anchor: FunctionComponent<Props> = ({ url, children, title }) => (
  <Link to={url} title={title}>
    {children}
  </Link>
)

export default Anchor
