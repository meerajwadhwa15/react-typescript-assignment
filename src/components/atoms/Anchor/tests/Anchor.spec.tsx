import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Anchor, { Props } from '../Anchor'

const props: Props = {
  url: '/:1',
  children: 'test',
  title: 'Test 1',
}

describe('Anchor Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(
      <BrowserRouter>
        <Anchor {...props} />
      </BrowserRouter>
    )
  })
})

describe('Anchor Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(
      <BrowserRouter>
        <Anchor {...props} />
      </BrowserRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
