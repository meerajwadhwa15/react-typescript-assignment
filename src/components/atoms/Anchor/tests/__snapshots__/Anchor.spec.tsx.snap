// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Snapshot testing Matches the snapshot 1`] = `
<a
  href="/:1"
  onClick={[Function]}
  title="Test 1"
>
  test
</a>
`;
